src/common/Configuration/Config.cpp

```diff
@@ -31,64 +32,65 @@ namespace
     std::vector<std::string> _args;
     std::unordered_map<std::string /*name*/, std::string /*value*/> _configOptions;
     std::mutex _configLock;
+    bool _usingDistConfig = false;
 
     // Check system configs like *server.conf*
     bool IsAppConfig(std::string_view fileName)
     {
-        size_t found = fileName.find_first_of("authserver.conf");
-        if (found != std::string::npos)
-        {
-            return true;
-        }
-
-        found = fileName.find_first_of("worldserver.conf");
-        if (found != std::string::npos)
-        {
-            return true;
-        }
+        size_t foundAuth = fileName.find("authserver.conf");
+        size_t foundWorld = fileName.find("worldserver.conf");
 
-        return false;
+        return foundAuth != std::string_view::npos || foundWorld != std::string_view::npos;
     }
 
     template<typename Format, typename... Args>
     inline void PrintError(std::string_view filename, Format&& fmt, Args&& ... args)
     {
-        std::string message = Acore::StringFormat(std::forward<Format>(fmt), std::forward<Args>(args)...);
+        std::string message = Acore::StringFormatFmt(std::forward<Format>(fmt), std::forward<Args>(args)...);
 
         if (IsAppConfig(filename))
         {
-            printf("%s\n", message.c_str());
+            fmt::print("{}\n", message);
         }
         else
         {
-            LOG_ERROR("server.loading", "%s", message.c_str());
+            FMT_LOG_ERROR("server.loading", message);
         }
     }
 
-    void AddKey(std::string const& optionName, std::string const& optionKey, bool replace = true)
+    void AddKey(std::string const& optionName, std::string const& optionKey, std::string_view fileName, bool isOptional = false)
     {
         auto const& itr = _configOptions.find(optionName);
-        if (itr != _configOptions.end())
+
+        // Check old option
+        if (isOptional && itr == _configOptions.end())
         {
-            if (!replace)
-            {
-                LOG_ERROR("server.loading", "> Config: Option '%s' is exist! Option key - '%s'", optionName.c_str(), itr->second.c_str());
-                return;
-            }
+            FMT_LOG_WARN("server.loading", "> Config::LoadFile: Found incorrect option '{}' in config file '{}'. Skip", optionName, fileName);
```

Comment:
```suggestion
            FMT_LOG_WARN("server.loading", "> Config::LoadFile: Found incorrect option '{}' in config file '{}'. Skip", optionName, fileName);
            
            if (!reload)
            {
               ABORT()
            }
```
NOTE: of course you also need to pass the "reload" flag to the function

We want to stop the server startup if there's a wrong conf. This can be very dangerous if AC updates the name of one conf and suddenly you have wrong behaviours in-game because you forgot to check the startup errors.

Reply:
Hm. good